tests/test_collection_get.py

```diff
@@ -131,6 +131,312 @@ def _insert_test_data(self, client, collection_name: str):
             client._server._execute(sql)
         
         return inserted_ids
+
+    def test_embedded_metadata_array_in_overlap(self):
+        """
+        Regression test for JSON array $in operator with embedded client.
+        Tests that tags=["ml","ai"] matches $in ["ml","python"] via JSON_OVERLAPS.
+        """
+        try:
+            import pylibseekdb  # noqa: F401
+        except ImportError:
+            pytest.skip("seekdb embedded package is not installed")
+
+        client = pyseekdb.Client(
+            path=SEEKDB_PATH,
+            database=SEEKDB_DATABASE
+        )
+
+        collection_name = f"test_tags_in_{int(time.time() * 1000)}"
+        collection = client.get_or_create_collection(
+            name=collection_name,
+            embedding_function=pyseekdb.DefaultEmbeddingFunction(),
+        )
+
+        try:
+            collection.add(
+                ids="id1",
+                documents="",
+                metadatas={"category": "AI", "score": 95, "tags": ["ml", "ai"], "version": 1},
+            )
+
+            result = collection.get(
+                where={"tags": {"$in": ["ml", "python"]}},
+                include=["metadatas", "ids"],
+            )
+
+            # Expect 1 row because tags array overlaps with the query list
+            assert result is not None
+            assert "ids" in result
+            assert len(result["ids"]) == 1
+            assert result["metadatas"][0].get("tags") == ["ml", "ai"]
+        finally:
+            try:
+                client.delete_collection(name=collection_name)
+            except Exception as cleanup_error:  # pragma: no cover
+                print(f"Warning: cleanup failed for {collection_name}: {cleanup_error}")
+    
+    def test_server_metadata_array_in_overlap(self):
```

Comment:
We can simplify the 3-mode testing, for example:
```python
def _test_metadata_array_in_overlap(self, client):
    # do something
def test_server_metadata_array_in_overlap(self):
    client = pyseekdb.Client(host=...)
    self._test_metadata_array_in_overlap(client)
def test_oceanbase_metadata_array_in_overlap(self):
    client = pyseekdb.Client(host=host, tenant=tenant)
    self._test_metadata_array_in_overlap(client)
```

Even though the code in this repo were written in this style, but it's ugly. We can do better now.